webapp/views.py

```diff
@@ -914,3 +914,21 @@ def thank_you():
     return flask.render_template(
         "thank-you.html", referrer=flask.request.args.get("referrer")
     )
+
+
+def subscription_centre():
+    leadId = flask.request.args.get("id")
```

Comment:
@carkod this URL param will come from the email, is there a way we can hide it from the browser URL bar in Flask?

Reply:
Hiding it from the browser wont solve the vulnerability, as the user will still see the id before reaching the page and be able to change it to access someone else's subscriptions

Reply:
Well, like Pete said. Since, this id comes from the browser request (client-side), so it is coming from the user, here flask already received the id.

Reply:
Sure - I'm not suggesting this solves the vulnerability, but if we can obscure it in any way, that's better than nothing.

Reply:
I have found another key within a user's info that appears to be random, I have to check with David if it is possible to use, but this could be a possible solution

Reply:
Even better if possible :crossed_fingers:

Reply:
I spoke to David and he confirmed it should be fine to use. I have just applied this change